File: fizzbuzz-test.php
<?php

function fizzbuzzWithConstraint($N) {
    for ($i = 1; $i <= $N; $i++) {
        if ($i % 3 === 0 && $i % 5 === 0) {
            echo "FizzBuzz\n";
        } elseif ($i % 3 === 0) {
            echo "Fizz\n";
        } elseif ($i % 5 === 0) {
            echo "Buzz\n";
        } else {
            echo $i . "\n";
        }
    }
}

// Input dari pengguna
$N = (int) readline("Masukkan nilai N: ");
fizzbuzzWithConstraint($N);

?>

File: fizzbuzz-test.go
package main

import (
	"fmt"
)

func fizzbuzzWithConstraint(N int) {
	for i := 1; i <= N; i++ {
		if i%3 == 0 && i%5 == 0 {
			fmt.Println("FizzBuzz")
		} else if i%3 == 0 {
			fmt.Println("Fizz")
		} else if i%5 == 0 {
			fmt.Println("Buzz")
		} else {
			fmt.Println(i)
		}
	}
}

func main() {
	// Input dari pengguna
	var N int
	fmt.Print("Masukkan nilai N: ")
	fmt.Scan(&N)
	fizzbuzzWithConstraint(N)
}
